### PR TITLE
Prepare for releasing v0.4.0

### PR DESCRIPTION
**Pull request by tenzen-y**
Mainly, I updated the release note.

Note: Blocked on #540 #543.

/assign @alculquicondor @terrytangyuan

### CODE REVIEW

**Comment by alculquicondor** [comment]
/hold
(to merge at the right time)

File: RELEASE.md
Comment (alculquicondor) [reply RELEASE.md:9]: Uhm... in theory, Kueue support is independent of this repo.

But maybe we can say:

`Support for suspend semantics. Third party controllers can leverage the suspend field to implement queuing and preemption for an MPIJob.`

File: RELEASE.md
Comment (alculquicondor) [inline RELEASE.md:7]:
```suggestion
* Bug fixes
  * Fix support for elastic Horovod.
```

**Review by terrytangyuan — APPROVED** [review]
Thanks!

/lgtm

File: RELEASE.md
Comment (alculquicondor) [inline RELEASE.md:3]: Also:
- Multi arch support #489

File: RELEASE.md
Comment (tenzen-y) [reply RELEASE.md:9]: I agree. Thank you for suggesting a great sentence!

Comment (tenzen-y) [reply RELEASE.md:11]: Done.

File: RELEASE.md
Comment (tenzen-y) [reply RELEASE.md]: Done.

File: RELEASE.md
Comment (tenzen-y) [reply RELEASE.md:3]: Done.

File: RELEASE.md
Comment (alculquicondor) [inline RELEASE.md:7]: nit: bug fixes at the end of the list

Comment (tenzen-y) [reply RELEASE.md]: Done.

**Comment by tenzen-y** [comment]
@alculquicondor Once #545 is merged into the master branch, can we merge this PR?

**Comment by alculquicondor** [comment]
Let me try to upload the images first.

**Comment by tenzen-y** [comment]
> Let me try to upload the images first.

Sure.

**Comment by tenzen-y** [comment]
Oh, conflicts are happening.

**Comment by tenzen-y** [comment]
> Oh, conflicts are happening.

Resolved.

File: Makefile
Comment (alculquicondor) [inline Makefile:20]: actually, we used `0.3.0` in the tag before, as opposed to `v0.3.0`

Comment (tenzen-y) [reply Makefile:20]: Upon confirmation, only `v0.2.3` has a `v` prefix.

**Comment by alculquicondor** [comment]
/lgtm

we can merge this right after the release is published in github

**Comment by alculquicondor** [comment]
or actually, before should be good

/hold cancel

**Comment by tenzen-y** [comment]
> or actually, before should be good

Previously (v0.3.0), we merged before cutting a new release.

https://github.com/kubeflow/mpi-operator/commits/v0.3.0

**Pull request merged**
